src/utils/gitUtils.ts

```diff
@@ -119,15 +134,7 @@ async function promptForCommit(repo: Repository, value?: string): Promise<void>
     const commitOptions: CommitOptions = { all: true };
 
     const commitMsg: string = await ext.ui.showInputBox({ prompt: commitPrompt, placeHolder: `${commitPrompt}..`, value });
-    try {
-        await repo.commit(commitMsg, commitOptions);
-    } catch (err) {
-        // ignore empty commit errors which will happen if a user initializes a blank folder or have changes in a nested git repo
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-        if (!/nothing to commit/.test(err.stdout)) {
-            throw new GitError(err);
-        }
-    }
+    await callWithGitErrorHandling(async () => await repo.commit(commitMsg, commitOptions));
```

Comment:
Maybe I missed something, but how come this single try/catch wasn't sufficient?

If you _do_ need all these extra calls to `callWithGitErrorHandling` - is there any way you can auto-magically add it to the git api for _all_ calls instead of manually adding it to each call like you're currently doing?

Reply:
It's because I was seeing issues that had `Failed to execute git`.  I noticed that CTI reported that error even when I had this try/catch in the build.  That made me feel a lack of confidence that all the git commands would just work, so I wanted to have better error handling for all of them.

I was thinking of a way I could automatically have all the commands have the error handling without doing what I'm doing. I'll try to think on it a bit more, might be able to do it if I get creative.

Reply:
I couldn't find a nice way to automagically wrap all the gitApi and repo commands so I opted to just use an error handler that makes this look a little cleaner.